### PR TITLE
[IOTDB-5852] Add some IoTConsensus and RatisConsensus properties

### PR DESCRIPTION
**Pull request by BUAAserein**
#### When testing IoTDB consensus layer performance, some properties of IoTCosensus and RatisConsensus are added. Considering their significant impact on performance, it is necessary to include them in the code repository.

Please see details in [jira](https://issues.apache.org/jira/browse/IOTDB-5852).

### CODE REVIEW

File: confignode/src/main/java/org/apache/iotdb/confignode/conf/ConfigNodeConfig.java [hunk 1]
```diff
@@ -237,7 +239,9 @@ public class ConfigNodeConfig {
   private long configNodeSimpleConsensusLogSegmentSizeMax = 24 * 1024 * 1024L;
 
   /** RatisConsensus protocol, flow control window for ratis grpc log appender */
-  private long dataRegionRatisGrpcFlowControlWindow = 4 * 1024 * 1024L;
+  private long dataRegionRatisGrpcFlowControlWindow = 16 * 1024 * 1024L;
```
Comment (SzyWilliam) [inline confignode/src/main/java/org/apache/iotdb/confignode/conf/ConfigNodeConfig.java:242]: Why change the default value?

Comment (SzyWilliam) [reply confignode/src/main/java/org/apache/iotdb/confignode/conf/ConfigNodeConfig.java:242]: I think we should change the default value after a thorough evaluation.
Maybe we can leave this PR clean and separate the changes of default value to another PR. We may need that PR when inviting community discussions.

File: consensus/src/main/java/org/apache/iotdb/consensus/ratis/metrics/IoTDBMetricRegistry.java [hunk 2]
```diff
@@ -74,6 +74,16 @@ public class IoTDBMetricRegistry implements RatisMetricRegistry {
   public static final String FOLLOWER_APPEND_ENTRIES_LATENCY = "follower_append_entry_latency";
   /** Time taken to process write requests from client. */
   public static final String RAFT_CLIENT_WRITE_REQUEST = "clientWriteRequest";
+  /** Applied index of state machine */
+  public static final String STATEMACHINE_APPLIED_INDEX_GAUGE = "appliedIndex";
+  /** Last log index which completely applied to the state machine */
+  public static final String STATEMACHINE_APPLY_COMPLETED_GAUGE = "applyCompletedIndex";
+  /** Next index of peer */
```
Comment (SzyWilliam) [inline consensus/src/main/java/org/apache/iotdb/consensus/ratis/metrics/IoTDBMetricRegistry.java:81]: Is these metrics not used yet?

File: confignode/src/main/java/org/apache/iotdb/confignode/conf/ConfigNodeConfig.java (hunk 1 above)
Comment (BUAAserein) [reply confignode/src/main/java/org/apache/iotdb/confignode/conf/ConfigNodeConfig.java:244]: Agree.

File: consensus/src/main/java/org/apache/iotdb/consensus/ratis/metrics/IoTDBMetricRegistry.java (hunk 2 above)
Comment (BUAAserein) [reply consensus/src/main/java/org/apache/iotdb/consensus/ratis/metrics/IoTDBMetricRegistry.java:81]: These metrics are already used in the RatisConsensus Metrics. Maybe we can put these changes in another PR.

**Review by SzyWilliam — APPROVED** [review]
+1 the change looks good. Thanks @BUAAserein a lot for working on this PR!

File: thrift-confignode/src/main/thrift/confignode.thrift [hunk 3]
```diff
@@ -77,8 +77,12 @@ struct TRatisConfig {
...
   26: required i64 firstElectionTimeoutMax
 
-  27: required i64 schemaRegionRatisLogMax
-  28: required i64 dataRegionRatisLogMax
+  29: required i64 schemaRegionRatisLogMax
+  30: required i64 dataRegionRatisLogMax
+
+  31: required i32 schemaGrpcLeaderOutstandingAppendsMax
+  32: required i32 dataGrpcLeaderOutstandingAppendsMax
+  33: required i32 dataLogForceSyncNum
```
Comment (OneSizeFitsQuorum) [inline thrift-confignode/src/main/thrift/confignode.thrift:85]: use schemaRegion and dataRegion

**Review by OneSizeFitsQuorum — COMMENTED** [review]
Please update the [docs](https://iotdb.apache.org/zh/UserGuide/Master/Reference/Common-Config-Manual.html).

File: thrift-confignode/src/main/thrift/confignode.thrift (hunk 3 above)
Comment (BUAAserein) [reply thrift-confignode/src/main/thrift/confignode.thrift]: ok

**Review by OneSizeFitsQuorum — APPROVED** [review]
LGTM

**Pull request merged**